core/husarnet/api_server/dashboard_api_proxy.cpp

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2024 Husarnet sp. z o.o.
+// Authors: listed in project_root/README.md
+// License: specified in project_root/LICENSE.txt
+#include "husarnet/api_server/dashboard_api_proxy.h"
+
+#include "husarnet/logging.h"
+
+bool DashboardApiProxy::isValid()
+{
+  return !apiAddress.empty();
+}
+
+void DashboardApiProxy::signAndForward(
+    const httplib::Request& req,
+    httplib::Response& res,
+    const std::string& path)
+{
+  nlohmann::json payload = nlohmann::json::parse(req.body, nullptr, false);
+  if(payload.is_discarded()) {
+    LOG_INFO("HTTP request body is not a valid JSON");
+    return;
+  }
+
+  auto pk = httplib::detail::base64_encode(identity->getPubkey());
+  auto sig = httplib::detail::base64_encode(identity->sign(payload.dump()));
```

Comment:
I kinda feel like… `dump`-ing the JSON is not an inherently "stable" operation. I.e. dictionaries can be dumped in different order (especially in two, distinct programming languages). I'd say it'd be better if payload was passed as a 1:1 string so checking the signature will be a stable (and straightforward) operation